sherpa/astro/utils/__init__.py

```diff
@@ -386,20 +387,14 @@ def _counts(data, lo, hi, func, *args):
 
 def _counts2d(data, reg, func, *args):
     old_mask = data.mask
-    coord = getattr(data, 'coord', None)
```

Comment:
I assume `coord` is here for the same reason that 
```
        if old_quality_filter is not None:
            data.quality_filter = old_quality_filter
```
is there for the 1d case. You never know if these notice calls mess something up. I've rarely if ever looked at the 2d case, so I don't know if that's possible.

Reply:
Yes, the 1D case has to worry about the quality array and the 2d case might want to worry about the coord setting (except that it doesn't actually, which is why it's being removed).